### ISSUE TITLE
Quality on mac is literally unusable

### ISSUE THREAD

**Issue by ishandutta2007** [atanunq/viu#30]
Not sure if I am missing something,

Installed like this:
`cargo install viu`

then opened an image like this:
`viu ~/Desktop/tenor.gif`

The quality is like this:

![Screen Shot 2019-12-14 at 1 47 34 PM](https://user-images.githubusercontent.com/2527354/70845759-69e8e180-1e78-11ea-8d9c-9f7d1f9e929e.png)

While  original gif is this

![tenor](https://user-images.githubusercontent.com/2527354/70845762-753c0d00-1e78-11ea-8bba-bf8b1961de3d.gif)

**Comment by atanunq** [comment]
Thank you for reporting. While I cannot test on a Mac myself, my guess would be this has to do with the terminal emulator you are using and/or how many colors does it support.
Sorry I cannot be more specific, hope this helps.

**Comment by ishandutta2007** [comment]
I guess you are on correct lines , maybe someone else can have more specific guidelines on how I need to customise my terminal to increase the resolution, I would request to keep the issue till then.

**Comment by rparrett** [comment]
Terminal.app doesn't support "true color."

It would be neat if viu could detect this, do color quantization and output ansi256.

**Comment by ishandutta2007** [comment]
> color quantization and output ansi256

sorry if I am acting too naive, I didn't quite get it, can you explain in code ?

**Comment by rparrett** [comment]
Apologies for not explaining well. I'm not an expert on the matter.

This gist explains the situation better than I could: https://gist.github.com/XVilka/8346728

And I happened upon this crate that converts 24-bit rgb "true color" to "ansi256" which may or may not be useful.

https://docs.rs/ansi_colours/1.0.1/ansi_colours

**Comment by atanunq** [comment]
Hello, all! Sorry for the very delayed response.

In branch [ansi256-support](https://github.com/atanunq/viu/tree/ansi256-support) I added a check to make sure truecolor is available before using it. This should hopefully fix the issues on Mac and other platforms where only 256 colors are supported.

It appeared to be working on the Mac I used but could use a few beta testers, if you don't mind, before releasing it.

Tagging @CaspianA1 because it might also be related to #33. And if not, I will have another look.
